lib/src/components/NxCombobox/__tests__/NxCombobox.test.tsx

```diff
@@ -0,0 +1,703 @@
+
+    expect(alertDropdownElement).toBeInTheDocument();
+
+    rerender(<NxCombobox { ...minimalProps } loading={true} />);
+    expect(alertDropdownElement).toBeInTheDocument();
+
+    rerender(<NxCombobox { ...minimalProps } loadError={'boo'} />);
+    expect(alertDropdownElement).toBeInTheDocument();
+
+    rerender(<NxCombobox { ...minimalProps } matches={ [{ id: '1', displayName: 'Foo' }] } />);
+    expect(alertDropdownElement).not.toBeInTheDocument();
+  });
+
+  it('sets aria-live on the alert dropdown to "polite"', function() {
+    expect(quickRender().getAllByRole('alert')[1]).toHaveAttribute('aria-live', 'polite');
+  });
+
+  it('sets aria-busy on the alert dropdown if loading is true', function() {
+    const { getAllByRole, rerender } = quickRender(),
+        alertDropdownElement = getAllByRole('alert')[1];
+
+    expect(alertDropdownElement).toHaveAttribute('aria-busy', 'false');
+
+    rerender(<NxCombobox { ...minimalProps } loading={true} />);
+    expect(alertDropdownElement).toHaveAttribute('aria-busy', 'true');
+  });
+
+  it('renders error and textcontent when the `loadError` prop is set', function() {
+    const { getAllByRole } = quickRender({ loadError: 'err' }),
+        errorElement = getAllByRole('alert')[1];
+
+    expect(errorElement).toBeInTheDocument();
+    expect(errorElement).toHaveTextContent('err');
+  });
+
+  it('fires onSearch with the search text when the retry button is clicked', async function() {
+    const user = userEvent.setup(),
+        onSearch = jest.fn(),
+        { getByRole } = quickRender({ value: 'f', loadError: 'err', onSearch }),
+        retryBtn = getByRole('button', { name: /retry/i });
+
+    expect(onSearch).not.toHaveBeenCalled();
+    await user.click(retryBtn);
+    expect(onSearch).toHaveBeenCalledWith('f');
+  });
+
+  it('fires onChange when the button with role option is clicked', async function() {
+    const user = userEvent.setup(),
+        onChange = jest.fn(),
+        { getAllByRole } = quickRender({
+          matches: [{ id: '1', displayName: 'Foo' }, { id: '2', displayName: 'Boo' }], onChange }),
+        optionBtns = getAllByRole('option');
+
+    await user.click(optionBtns[1]);
+    expect(onChange).toHaveBeenCalledWith('Boo');
+  });
+
+  it('renders empty message if there are no results', function() {
+    const { getAllByRole } = quickRender(),
+        emptyMessage = getAllByRole('alert')[1];
+
+    expect(emptyMessage).toBeInTheDocument();
+    expect(emptyMessage).toHaveTextContent('No Results Found');
+  });
+
+  it('sets the empty message from the `emptyMessage` prop', function() {
+    const { getAllByRole } = quickRender({ emptyMessage: 'asdfasdf' }),
+        emptyMessage = getAllByRole('alert')[1];
+
+    expect(emptyMessage).toBeInTheDocument();
+    expect(emptyMessage).toHaveTextContent('asdfasdf');
+  });
+
+  it('calls onSearch with the current trimmed search text if focus enters the component from elsewhere on the page ' +
```

Comment:
We got rid of the trimming.  It doesn't look like this test was really checking trimming, but its name should be updated